core/src/main/scala/dagr/core/execsystem/TaskTracker.scala

```diff
@@ -70,52 +70,55 @@ trait TaskTracker extends TaskManagerLike with LazyLogging {
     addTask(task=task, enclosingNode=None, ignoreExists=false)
   }
 
+  private def addTaskNoChecking(task: Task, enclosingNode: Option[GraphNode] = None): TaskId = {
+    // set the task id
+    val id = yieldAndThen(nextId) {nextId += 1}
+    // set the task info
+    require(task._taskInfo.isEmpty) // should not have any info!
+    val info = new TaskExecutionInfo(
+      task=task,
+      taskId=id,
+      status=UNKNOWN,
+      script=scriptPathFor(task=task, id=id, attemptIndex=1),
+      logFile=logPathFor(task=task, id=id, attemptIndex=1),
+      submissionDate=Some(Instant.now())
+    )
+    task._taskInfo = Some(info)
+
+    // create the graph node
+    val node = predecessorsOf(task=task) match {
+      case None => new GraphNode(task=task, predecessorNodes=Nil, state=GraphNodeState.ORPHAN, enclosingNode=enclosingNode)
+      case Some(predecessors) => new GraphNode(task=task, predecessorNodes=predecessors, enclosingNode=enclosingNode)
+    }
+
+    // update the lookups
+    idToTask.put(id, task)
+    idToNode.put(id, node)
+
+    id
+  }
+
   /** Adds a task to be managed
-    *
-    * Throws an [[IllegalArgumentException]] if a cycle was found after logging each strongly connected component with
-    * a cycle in the graph.
-    *
-    * @param ignoreExists true if we just return the task id for already added tasks, false if we are to throw an [[IllegalArgumentException]]
-    * @param task the given task.
-    * @return the task identifier.
-    */
+  *
+  * Throws an [[IllegalArgumentException]] if a cycle was found after logging each strongly connected component with
+  * a cycle in the graph.
+  *
+  * @param ignoreExists true if we just return the task id for already added tasks, false if we are to throw an [[IllegalArgumentException]]
+  * @param task the given task.
+  * @return the task identifier.
+  */
   protected[execsystem] def addTask(task: Task, enclosingNode: Option[GraphNode], ignoreExists: Boolean = false): TaskId = {
```

Comment:
Should we just eliminate this method (and force callers to wrap individual tasks in `Seq()`) so as to encourage use of the new `addTasks()`?